src/playlist-maker.js

```diff
@@ -71,6 +96,14 @@ var playlistMaker = function(player, plist) {
     playlist.currentItem(0);
   }
 
+  player.on('loadstart', function() {
+    var currentSrc = player.currentSrc();
+    if (!isInSources(list, currentSrc)) {
```

Comment:
looks like this is causing issues with ads and playlists because the ads gets loaded and it isn't in the sources, so, we think it's a non-playlist item.